apps/extension/src/background/approvals/messages.ts

```diff
@@ -147,7 +147,8 @@ export class ConnectInterfaceResponseMsg extends Message<void> {
 
   constructor(
     public readonly interfaceOrigin: string,
-    public readonly allowConnection: boolean
+    public readonly allowConnection: boolean,
+    public readonly chainId?: string
```

Comment:
I was hoping to make it optional while we transition to the new Namadillo, however, since the Signer class changed, it breaks the old Namadillo anyways :D I can easily make it required, will just need to update the approval tests!

Reply:
@euharrison So, to get this to work as a required param, we'll need to update `@namada/integrations` (and the Faucet app) which I was hoping to avoid in this PR. There is the case of the Faucet where a chain ID isn't needed, as all it does is load accounts, so I'm wondering how we should handle that? I think for now, it shouldn't hurt to leave it as optional, and _only_ set the Network setting if it is provided.